src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py

```diff
@@ -28,13 +28,22 @@ def _process_request_recording(request):
                 key in SNOWFLAKE_REQUEST_ID_STRINGS
                 or key in SNOWFLAKE_DB_RELATED_FIELDS_IN_QUERY
             ):
-                request.uri = request.uri.replace(value, VOID_STRING)
+                if request.uri.find(value) ==-1:
+                    request.uri = request.uri.replace(urllib.parse.quote_plus(value), VOID_STRING)
+                else:
+                    request.uri = request.uri.replace(value, VOID_STRING)
 
         # scrub snowflake account information
         if request.host.endswith(".snowflakecomputing.com"):
             account = request.host.split(".snowflakecomputing.com")[0]
             request.uri = request.uri.replace(account, VOID_STRING)
 
+        if request.host.endswith(".amazonaws.com"):
+            account = request.host.split(".amazonaws.com")[0]
+            code = request.uri.split("/results/")[1].split("/")[0]
+            request.uri = request.uri.replace(account, VOID_STRING)
+            request.uri = request.uri.replace(code, VOID_STRING)
```

Comment:
what about gcp and azure?

Reply:
let's create a JIRA and add a comment in the code that azure and gcp will handled separately